api/bucket-handlers.go

```diff
@@ -315,8 +315,33 @@ func (api ObjectAPIHandlers) DeleteMultipleObjectsHandler(w http.ResponseWriter,
 	var deletedObjects []ObjectIdentifier
 	// Loop through all the objects and delete them sequentially.
 	for _, object := range deleteObjects.Objects {
-		result, err := api.ObjectAPI.DeleteObject(bucket, object.ObjectName,
-			object.VersionId, credential)
+		// TODO: Delete all objects in one transaction
+		ctx.ObjectName = object.ObjectName
+		ctx.VersionId = object.VersionId
+		ctx.ObjectInfo, err = api.ObjectAPI.GetObjectInfo(bucket, object.ObjectName, object.VersionId, credential)
+		if err != nil {
+			if err == ErrNoSuchKey {
+				continue
+			}
+			logger.Error("Unable to delete object:", err)
+			apiErrorCode, ok := err.(ApiErrorCode)
+			if ok {
+				deleteErrors = append(deleteErrors, DeleteError{
+					Code:      ErrorCodeResponse[apiErrorCode].AwsErrorCode,
+					Message:   ErrorCodeResponse[apiErrorCode].Description,
+					Key:       object.ObjectName,
+					VersionId: object.VersionId,
+				})
+			} else {
+				deleteErrors = append(deleteErrors, DeleteError{
+					Code:      "InternalError",
+					Message:   "We encountered an internal error, please try again.",
+					Key:       object.ObjectName,
+					VersionId: object.VersionId,
+				})
+			}
+		}
+		result, err := api.ObjectAPI.DeleteObject(ctx, credential)
```

Comment:
Just pass down object name and version, don't rely on context do parameter passing. The usage pattern of context should be "write once, read everywhere".

Also, is this change really making operations faster?

Reply:
I will fix it in next pr